### PR TITLE
(SIMP 7227) Add nmcli support for DNS servers

### PR DESCRIPTION
**Pull request by kendall-moore**
* Use connection name instead of device name as this is more user-controlled
* Improve Network Manager validation
* Update REFERENCE to reflect new class parameters

SIMP-7227 #close

### CODE REVIEW

**Comment by kendall-moore** [comment]
Local validation of acceptance tests: 
image

GitLab CI Pipeline:
https://gitlab.com/simp/pupmod-simp-resolv/-/pipelines/164081299

**Comment by op-ct** [comment]
@kendall-moore is there a reason this PR is coming from a branch in the upstream repo instead of the [kendall-moore/pupmod-simp-resolv](https://github.com/kendall-moore/pupmod-simp-resolv) fork?

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,8 @@
+* Tue Jul 07 2020 Kendall Moore <kendall.moore@onyxpoint.com> - 0.4.1-0
```
Comment (op-ct) [inline CHANGELOG:1]: We'll need to bump the module version in metadata.json before release.  I suggest 0.5.0 (details in manifests/init.pp)

File: manifests/init.pp [hunk 2]
```diff
@@ -71,48 +71,57 @@
   Boolean                    $named_autoconf            = true,
   Boolean                    $caching                   = true,
   Boolean                    $use_nmcli                 = false,
-  Optional[String]           $nmcli_device_name         = undef,
+  Optional[String]           $nmcli_connection_name     = undef,
```
Comment (op-ct) [inline manifests/init.pp:74]: This parameter name change requires a version bump.

It doesn't need to be a major bump―the resolve module is major version 0 (which SemVer considers it [initial development](https://semver.org/#spec-item-4).  However, it's an API-breaking change to a published module (0.4.0) and introduces new functionality, so I'd suggest 0.5.0.

**Comment by kendall-moore** [comment]
Closing to re-open from personal fork instead of base repo.

**Pull request closed without merging**